_posts/2017-11-13-setup-monit-notifications-on-slack.md

```diff
@@ -0,0 +1,112 @@
+
+The second part requires you to create a <a href="https://en.wikipedia.org/wiki/Bash_(Unix_shell)"> Bash </a> script that will post Slack messages when you run it.
+This is an example of how this file should look like:
+
+```
+URL="https://hooks.slack.com/services/T7W3HFJSO/B7XLPmQAZ/aviXjd5cKdzvzUjfWbCn8oEqr" # Slack Webhook URL
+
+PAYLOAD="{
+  \"attachments\": [
+    {
+      \"title\": \"$PROCESS was restarted\",
+      \"color\": \"warning\",
+      \"mrkdwn_in\": [\"text\"],
+      \"fields\": [
+        { \"title\": \"Date\", \"value\": \"$MONIT_DATE\", \"short\": true },
+        { \"title\": \"Host\", \"value\": \"$MONIT_HOST\", \"short\": true }
+      ]
+    }
+  ]
+}"
+
+curl -s -X POST --data-urlencode "payload=$PAYLOAD" $URL
+```
+
+If you want to see how to customize the Slack messages you can take a look at the [official documentation](https://api.slack.com/incoming-webhooks#sending_messages).
+
+Before run the script we should add some permissions (I saved it as `slack-webhook.sh` in `/usr/local/bin/`)
+
+```
+$ sudo chmod u+x /usr/local/bin/slack-webhook.sh
+```
+
+Now let's run it to see if it's working
+
+```
+$ /usr/local/bin/slack-webhook.sh
+```
+
+If correct, you will see the message in your Slack channel.
+
+<img src="/blog/assets/images/monit-message.png" alt="Monit message">
+
+It looks a bit empty because we didn't set any variables for the message to use (e.g. `$PROCESS`).
+
+### Setup Monit
+
+You can see if you already have installed Monit by running
+
+```
+$ monit --version
+This is Monit version 5.16
+```
+
+If not you can install it with
+```
+$ sudo apt install monit -y
+```
+
+Now that you have Monit installed, you can add the configuration for your process on `/etc/monit/conf.d`
+
+
+```
+check process mosquitto with pidfile /var/run/mosquitto.pid
+  start program = "/etc/init.d/mosquitto start"
+  stop program = "/etc/init.d/mosquitto stop"
+
+  if changed pid then exec "/bin/bash -c 'PROCESS=Mosquitto /usr/local/bin/slack-webhook.sh'"
+  if 1 restart within 1 cycle then exec "/bin/bash -c 'PROCESS=Mosquitto /usr/local/bin/slack-webhook.sh'"
+```
+
+This configuration will restart the process in case you stop it manually or if it gets stop by itself. And when that happens it will call the `slack-webhook` script to publish a message in the Slack channel.
+You can see a bunch of real-world configuration examples on [this link](https://mmonit.com/wiki/Monit/ConfigurationExamples).
+
+Don't forget to restart Monit after make changes in the configuration.
+
+```
+$ monit reload
+Reinitializing monit daemon
+```
+
+You can easily test this by manually killing the process.
+
+```
+$ ps -aux | grep mosquitto
```

Comment:
@lubc I'd use `ps aux | grep mosquitto` instead of `ps -aux | grep mosquitto`, since `ps aux` is compatible across Linux and OSX and behaves the same way, and `ps -aux` is exclusive to Linux.

Reply:
ah, cool!